packages/dds/tree/src/shared-tree/treeAlpha.ts

```diff
@@ -339,6 +353,44 @@ export interface TreeAlpha {
 	 * Otherwise, this returns the key of the field that it is under (a `string`).
 	 */
 	key2(node: TreeNode): string | number | undefined;
+
+	/**
+	 * Gets the child of the given node with the given property key if a child exists under that key.
+	 *
+	 * @remarks Unknown optional fields of Object nodes cannot be accessed using this method.
```

Comment:
Do we explicitly define "unknown optional fields" somewhere? I imagine this refers to optional feels that have never been set and thus their key might not even exist in the Object node yet. If we define the term and can point to the definition, awesome, otherwise I'd suggest a more verbose doc.

Reply:
Found [Noah's comment](https://github.com/microsoft/FluidFramework/pull/24723/files#r2124792994) that applies here too.